src/sentry/eventstore/filter.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import absolute_import
+
+
+class Filter(object):
+    conditions = []
+    filter_keys = {}
+    start = None
+    end = None
+
+    def __init__(self, start=None, end=None, conditions=None, filter_keys=None):
+        self.start = start
+        self.end = end
+
+        if conditions:
+            self.conditions = conditions
+
+        if filter_keys:
+            self.filter_keys = filter_keys
+
+    def to_snuba_args(self):
```

Comment:
There are a few cleaner way to provide the Snuba specific logic to generate the condition:
1)put the method in the SnubaEventStorage. It takes the abstract filter and returns snuba condition. Should there ever be a different implementaiton, it would do the same
2) Have a SnubaQuery class that is fed a Filter and it is able to build the parameters for Snuba raw query.

While this may seem overkill now, it makes the design more clear because it splits responsibilties properly making them more visible to whoever reads the code.
On the other hand, relying on a class to provide a method per implementation blows up quickly if we have more than one implementation.